### PR TITLE
CONN-2437 import wizard part2

### PR DESCRIPTION
**Pull request by trantang**
next four-tabs

### CODE REVIEW

File: Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/CertificateUtil.java [hunk 1]
```diff
@@ -245,4 +250,22 @@ public static DataHandler getDataHandlerFrom(Certificate cert) throws Certificat
...
 
+    public static DataHandler getDataHandlerForPem(byte[] bytes) {
+        return new DataHandler(new ByteArrayDataSource(bytes, MIME_TEXT_PLAIN));
+    }
+
+    public static PrivateKeyEntry getPrivateKeyEntry(KeyStore keystore, String alias, String password)
+        throws CertificateManagerException {
+
+        if (StringUtils.isBlank(alias) || StringUtils.isBlank(password)) {
+            throw new CertificateManagerException("Empty alias or password for the keystore.");
```
Comment (plobre) [inline Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/opensaml/CertificateUtil.java:261]: Is it a requirement that a keystore must have a password to use CONNECT? You can have a blank password for a JKS.

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 2]
```diff
@@ -684,20 +732,167 @@ public void cancelImportWizard() {
...
+    public Map<String, String> getCaLinks() {
+        if (null == caLinks) {
+            loadCaProperties();
+        }
+        return caLinks;
+    }
+
+    public void importToKeystore() {
+        if (null == uploadedFileServer || StringUtils.isBlank(alias)) {
+            HelperUtil.addMessageError(null, "Alias and Server upload are required.");
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: Make message more clear

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 3]
```diff
@@ -684,20 +732,167 @@ public void cancelImportWizard() {
...
+        uploadedFileRoot = event.getFile();
+        if (uploadedFileRoot != null) {
+            certRoot = service.createCertificate(uploadedFileRoot.getContents());
+            validateCertificate(certRoot);
+        }
+    }
+
+    public void importToTruststore() {
+        if (MapUtils.isEmpty(listIntermediate) || null == uploadedFileRoot || StringUtils.isBlank(alias)) {
+            HelperUtil.addMessageError(null, "Alias, Intermediate and Root are required.");
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: Make message more clear

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 4]
```diff
@@ -569,13 +599,126 @@ private X509Certificate generateCertificate(KeyPair keypair, String dn, int days
...
+            }
+
+            CoreHelpUtils.saveJksTo(getTempTruststore(), System.getProperty(CertificateManagerImpl.TRUST_STORE_PASSWORD_KEY),
+                FILE_JKS_CACERTS_NEW);
+        }catch(CertificateManagerException | KeyStoreException | UtilException | CertificateEncodingException e){
+            LOG.error("error while importing to the temporary truststore: {}",e.getLocalizedMessage(), e);
+            return buildSimpleResponse(false, "error while importing to the temporary truststore.");
+        }
+
+        return buildSimpleResponse(true, "import temporary truststore successful.");
```
Comment (TabassumJafri) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: Make message more clear

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml [hunk 5]
```diff
@@ -471,21 +482,268 @@
...
+                            </p:tab>
+                            <p:tab id="tabIdCaAuthorityInfo" title="#{certificateBean.tabTitles[3]}" closable="false">
+
+                                <div class="form-block center-block">
+                                    <h:form id="formCaAuthorityInfo" class="form form-horizontal">
+                                        <p:messages id="errorMsgsCaAuthorityInfo" closable="true">
+                                            <p:autoUpdate />
+                                        </p:messages>
+                                        <div class="content">
+                                            <h:outputText>Go to your selected CA Authority website and enter the authorization code and the reference number.
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml]: Update message like this: "Go to your selected CA Authority website and enter the authorization code and reference number. Follow instructions to generate certificate. Copy the generated certificate  and ??"

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Review by TabassumJafri — APPROVED** [review]

**Review by ptambellini01 — APPROVED** [review]

**Review by plobre — APPROVED** [review]

**Pull request merged**